### PR TITLE
[pkg/otlp] Add support for OTLP metrics

### PR DESCRIPTION
**Pull request by mx-psi**
### What does this PR do?

- Add `TracesEnabled` OTLP pipeline configuration.
- Add a `serializerexporter` that translates and sends OTLP metrics to a metrics serializer.
- Add support for OTLP metrics to the OTLP pipeline.

### Motivation

DataDog/architecture#892

### Additional Notes

Metrics are enabled by default since this is still experimental. Should they be?

### Describe how to test your changes

Set up the OTLP pipeline by providing a port. Check that:
- [ ] if sent via an instrumentation library, timeseries metrics (e.g. OTLP Gauges or Sums) are correctly sent to the backend and that there are no logs.
- [ ] OTLP Histograms are sent correctly: if you send a `test.histogram` metric it will come up as `test.histogram` in the backend and have the usual functions available, with no extra `.count` or `.sum` metrics.
- [ ] The endpoint can receive metrics and traces at the same time.
- [ ] Both gRPC and HTTP work correctly.

### Checklist

- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] The `need-change/operator` and `need-change/helm` labels has been applied if applicable.
- [x] The appropriate `team/..` label has been applied, if known.
- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] The [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated if applicable.

Note: Adding GitHub labels is only possible for contributors with write access.

### CODE REVIEW

**Review by KSerrania — APPROVED** [review]
LGTM

File: pkg/otlp/config_test.go [hunk 1]
```diff
@@ -16,8 +16,11 @@ import (
```
Comment (KSerrania) [inline pkg/otlp/config_test.go:16]: `"fmt"` needs to be added to the import list here

Comment (mx-psi) [reply pkg/otlp/config_test.go:16]: Done

File: pkg/otlp/config_parser_test.go [hunk 2]
```diff
@@ -48,37 +54,50 @@ service:
 `,
 		},
 		{
-			name: "only HTTP",
+			name: "only HTTP, metrics and traces",
```
[the remark this answers is not in the document]
Comment (mx-psi) [reply pkg/otlp/config_parser_test.go:57]: Done

**Review by albertvaka — APPROVED** [review]
LGTM. I find a bit counter-intuitive that the way to enable OTLP is by setting `otlp.http_port`/`otlp.grpc_port`, given that now there are options called `otlp.metrics_enabled` and `otlp.traces_enabled`, but I'm not sure how to make it better. Maybe by default the `*_port` options should have a value (if there's a default value that makes sense to use) and `_enabled` be `false`? Then you would enable OTLP by setting the `*_enabled` options to `true`, which is what one might expect.

File: pkg/otlp/config_parser.go [hunk 3]
```diff
@@ -34,41 +34,83 @@ service:
       exporters: [otlp]
 `
 
+// metricsConfig is the metrics OTLP pipeline configuration.
+// TODO (AP-1254): Set service-level configuration when available.
+const metricsConfig string = `
```
Comment (djmitche) [inline pkg/otlp/config_parser.go:39]:
```suggestion
const defaultMetricsConfig string = `
```
if I read the code right, this is used as the default and user config is merged over it

**Review by djmitche — APPROVED** [review]
Looks good from my limited perspective.  If you'd like someone from the team with more understanding of the OTLP work, please flag the team for review again.

File: pkg/otlp/config_parser.go (hunk 3 above)
Comment (mx-psi) [reply pkg/otlp/config_parser.go]: You are right, renamed it!

**Comment by mx-psi** [comment]
> Maybe by default the *_port options should have a value (if there's a default value that makes sense to use) and _enabled be false? Then you would enable OTLP by setting the *_enabled options to true, which is what one might expect.

I think that makes sense, but I don't want to break config for existing OTLP traces users twice (one now with that change and another one for the stable configuration). We can definitely do something like that for the stable configuration, for now `metrics_enabled` and `traces_enabled` will work more as failsafes in case something is broken

**Comment by mx-psi** [comment]
Merging since the failing test is a known unrelated issue

**Pull request merged**